app/controllers/provider_interface/candidates_controller.rb

```diff
@@ -21,11 +21,10 @@ def impersonate
   private
 
     def verify_provider_association(candidate:, providers:)
-      provider_ids_from_candidate = candidate.application_forms
-                                              .map(&:application_choices).flatten
-                                              .map(&:provider).map(&:id).uniq
+      application_choices = candidate.application_forms.map(&:application_choices).flatten
+      providers_from_applications = application_choices.map(&:provider) + application_choices.map(&:accredited_provider)
```

Comment:
Could we utilise [associated_providers](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/models/application_choice.rb#L155)  which extracts both provider and accredited provider? 
```
providers_from_applications  = application_choices.flat_map(&:associated_providers).uniq
```